src/main/kotlin/com/exactpro/th2/codec/api/impl/ThreadSafeCodec.kt

```diff
@@ -21,15 +21,24 @@ import com.exactpro.th2.codec.api.IPipelineCodecFactory
 import com.exactpro.th2.codec.api.IPipelineCodecSettings
 import com.exactpro.th2.codec.api.IReportingContext
 import com.exactpro.th2.common.grpc.MessageGroup
+import mu.KotlinLogging
 import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit.MINUTES
+import java.util.concurrent.TimeUnit.SECONDS
 import javax.annotation.concurrent.ThreadSafe
 
 @ThreadSafe
 class ThreadSafeCodec(
     private val codecFactory: IPipelineCodecFactory,
     private val codecSettings: IPipelineCodecSettings?
 ) : IPipelineCodec {
-    private val instances = ConcurrentHashMap<Long, IPipelineCodec>()
+    private val logger = KotlinLogging.logger {}
+    private val instances = ConcurrentHashMap<Thread, IPipelineCodec>()
+
+    private val executor = Executors.newSingleThreadScheduledExecutor().apply {
```

Comment:
We already have an executor for storing events. Maybe we can reuse it to avoid extra threads being created?

Reply:
Sorry, where is it? I can't seem to find it

Reply:
Sorry, I was wrong. This is true only for codec-sailfish project